src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java

```diff
@@ -0,0 +1,76 @@
+package it.pagopa.ecommerce.commons.client;
+
+import it.pagopa.ecommerce.commons.exceptions.NpgResponseException;
+import it.pagopa.generated.ecommerce.npg.v1.api.HostedFieldsApi;
+import it.pagopa.generated.ecommerce.npg.v1.api.PaymentServicesApi;
+import it.pagopa.generated.ecommerce.npg.v1.dto.CreateHostedOrderRequestDto;
+import it.pagopa.generated.ecommerce.npg.v1.dto.FieldsDto;
+import it.pagopa.generated.ecommerce.npg.v1.dto.PostMessageDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.util.UriBuilder;
+import reactor.core.publisher.Mono;
+
+import javax.validation.constraints.NotNull;
+import java.util.UUID;
+
+/**
+ * <p>
+ * NpgClient instance to communicate with npg.
+ * </p>
+ */
+@Slf4j
+public class NpgClient {
+
+    /**
+     * The npg Api
+     */
+    private final PaymentServicesApi paymentServicesApi;
+
+    /**
+     * Instantiate a npg-client to establish communication via the npg api
+     *
+     * @param paymentServicesApi the api
+     * @param npgKey             the api key
+     */
+    public NpgClient(
+            @NotNull PaymentServicesApi paymentServicesApi,
+            @NotNull String npgKey
+    ) {
+        this.paymentServicesApi = paymentServicesApi;
+        this.paymentServicesApi.getApiClient().setApiKey(npgKey);
+    }
+
+    /**
+     * method to invoke the orders/build api
+     *
+     * @param createHostedOrderRequestDto the request to create the session
+     * @param correlationId               the unique id to identify the rest api
+     *                                    invocation
+     * @return An object containing sessionId, sessionToken and the fields list to
+     *         show on the client-side
+     */
+    public Mono<FieldsDto> buildOrders(
+                                       UUID correlationId,
+                                       CreateHostedOrderRequestDto createHostedOrderRequestDto
```

Comment:
`CreateHostedOrderRequestDto` has a weak interface (e.g. `amount` is a string, not a number, currency symbols can be arbitrary strings, and some action-params combinations are not permitted by the API but constructible), can we either make a custom data structure that correctly encodes invariants about the input request or limit the interface of this method?

Reply:
I leaved this method and added a new one with explicit parameters and hidden dto building behind the scenes.
Yours is a good point, but for this first phase of develop a direct api with the dto could be more useful and practice to make an hot-fix or a simple test, while we will tune the paramenters of the request.